lib/ansible/module_utils/redfish_utils.py

```diff
@@ -826,6 +826,45 @@ def get_fan_inventory(self):
         result["entries"] = fan_results
         return result
 
+    def get_chassis_power(self):
+        result = {}
+        key = "Power"
+
+        # Get these entries, but does not fail if not found
+        properties = ['Name', 'PowerAllocatedWatts',
+                      'PowerAvailableWatts', 'PowerCapacityWatts',
+                      'PowerConsumedWatts', 'PowerMetrics',
+                      'PowerRequestedWatts', 'RelatedItem', 'Status']
+
+        chassis_power_results = []
+
+        # Go through list
+        for chassis_uri in self.chassis_uri_list:
+            chassis_power_result = {}
+
+            response = self.get_request(self.root_uri + chassis_uri)
+            if response['ret'] is False:
+                return response
+            result['ret'] = True
+            data = response['data']
+            if key in data:
+                response = self.get_request(self.root_uri + chassis_uri +
+                                            "/" + key)
+                data = response['data']
+                if 'PowerControl' in data:
+                    data = data['PowerControl'][0]
```

Comment:
After testing with some additional systems/mockups, I encountered an error (`IndexError: list index out of range`) on this line with a `Power` resource that had an empty `PowerControl` array. An empty array is spec conformant, so it would be good to avoid the `IndexError`. Suggest an additional test for `len(data['PowerControl']) > 0` before trying to read the element.